src/newusers.c

```diff
@@ -350,22 +350,20 @@ static int get_user_id (const char *uid, uid_t *nuid) {
 			const struct passwd *pwd;
 			/* local, no need for xgetpwnam */
 			pwd = getpwnam (uid);
-			if (NULL == pwd) {
+			if (pwd == NULL)
```

Comment:
No, while I agree this does look odd to me as well, the original intent - making sure that if there is accidental assignment (= instead of ==) still is in my eyes noble.  Let's leave cases where NULL == X instead of X == NULL.

Reply:
I guessed that was the intent of the code.  While it was reasonable some decades ago, to avoid accidents, compilers know that something's wrong with assignments in conditionals and issue a warning:

```c
alx@asus5775:~/tmp$ cat assign.c 
int f(int x)
{
	if (x = 3)
		return 5;
	return 0;
}
```
```
alx@asus5775:~/tmp$ cc -Wall -Wextra assign.c 
assign.c: In function ‘f’:
assign.c:3:13: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
    3 |         if (x = 3)
      |             ^
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu/Scrt1.o: in function `_start':
(.text+0x17): undefined reference to `main'
collect2: error: ld returned 1 exit status
```

So, these days I think it's preferable to write readable (non-ugly) code, and trust in the compiler (and/or static analyzers) for those warnings.  I wouldn't go so far as rewriting all of those, but if I'm editing a conditional block for some reason, I find it enough of an excuse to rewrite the condition itself.

What do you think?

Reply:
Heh, ok, I'm good with that.